### PR TITLE
Fix babel support for legacy browsers

### LINKED ISSUE #2046: webOS: Use of const in strict mode (webcomponents?)
**Describe The Bug**

Crash on splash screen;
`Uncaught SyntaxError: Use of const in strict mode.` in bundle.js:220
Looks like it may be webcomponents – 220 is the `!function` line.
```js
/**
 * @license
 * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
 * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
 * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
 * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
 * Code distributed by Google as part of the polymer project is also
 * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 */
!function(){window.WebComponents=window.WebComponents||{flags:{}};var e=document.querySelector('script[src*="webcomponents-lite.js"]')…
```

**Steps To Reproduce**

1. Open Jellyfin in a sufficiently old browser (webOS 3.0 emulator in this particular case)
4. See error in console

**Logs**

image

**System (please complete the following information):**
 - Platform: webOS TV Emulator 3.0.0
 - Browser: [Chromium 38](http://webostv.developer.lge.com/discover/specifications/web-engine/)
 - Jellyfin Version: jellyfin-web 20201107.21-unstable

### PR DESCRIPTION
**Pull request by dmitrylyzo**
**Changes**
- Setup webpack to use browserslist.
- Add polyfill for dynamic import of external module.
- Re-add polyfills to fix `slideshow` and `scrollManager`.

**Issues**
Fixes #2046

### CODE REVIEW

**Comment by dmitrylyzo** [comment]
Need (?) to be transpiled (to remove `const`):
- `serviceworker.js`
- `pdf.worker.js`
- `worker-bundle.js`

`SubtitleOctopus` doesn't seem to work because it's not being imported anywhere.

**Comment by dmitrylyzo** [comment]
> `SubtitleOctopus` doesn't seem to work because it's not being imported anywhere.

Babel breaks it and other 2 modules, so I added an extra `babel-loader` block.

`pdfPlayer` seems to work.

On webOS 1.2, `SubtitlesOctopus` cannot load `.../web/libraries/subtitles-octopus-worker-legacy.js.mem` (server returns 404), but file is there.

**Comment by oddstr13** [comment]
Tested using the webOS 3 emulator
Not everything works yet, but at least Jellyfin loads now.

#### Books
 - pdf: `Uncaught SyntaxError: Unexpected token ( pdf.worker.js:2` (Same error that prevented Jellyfin completely from loading without this PR)
 - epub: Seem to just not load. White screen with loading circle for a little while, then all the controls disappear.
  I can resume browsing Jellyfin by using the back button.

#### Video
Looks like something's wrong with the playback profiles?
Audio only, white background behind the OSD.
```
media element error: 4  htmlVideoPlayer-plugin.bundle.js:1
playbackmanager playback error type: medianotsupported main.bundle.js:2
```

```
Unhandled promise rejection undefined main.bundle.js:2
```
Edit: I don't know what's going on any more. Sometimes it works, sometimes it doesn't…

**Comment by dmitrylyzo** [comment]
> Tested using the webOS 3 emulator

imo, webOS 3 emulator is buggiest one.

> I don't know what's going on any more. Sometimes it works, sometimes it doesn't…

In webOS 1.2 and webOS 3 emulators, DirectPlay video works, Transcode - doesn't (black screen, no progress). I have this behavior since I started working with Jellyfin on webOS.

Current problems:
1. ASS doesn't work because server doesn't return file `libraries/subtitles-octopus-worker-legacy.js.mem` (_as I said above_).
2. I don't know how to transpile workers (using webpack config):
`worker-bundle.js` (`libarchive.js`)
`pdf.worker.js` (`pdfjs-dist`)
`serviceworker.js`

**Comment by dmitrylyzo** [comment]
I excluded `SubtitlesOctopus` from babel because `lastRendered` is global which cannot be set in strict mode.
https://github.com/jellyfin/JavascriptSubtitlesOctopus/blob/58e9a3f1a7f7883556ee002545f445a430120639/src/subtitles-octopus.js#L331

It seems that this `const` is guarded by `try/catch`, so babel is not needed :crossed_fingers:
https://github.com/jellyfin/JavascriptSubtitlesOctopus/blob/58e9a3f1a7f7883556ee002545f445a430120639/src/subtitles-octopus.js#L6

The rest of issues are left for next time.

File: webpack.common.js [hunk 1]
```diff
@@ -91,11 +90,20 @@ module.exports = {
...
+            },
+            /* modules that Babel breaks when transforming to ESM */
+            {
+                test: /node_modules[\\/](pdfjs-dist|xmldom)[\\/].*\.js$/,
                 use: [{
                     loader: 'babel-loader',
                     options: {
-                        presets: packageConfig.babel.presets
+                        plugins: [
+                            '@babel/transform-modules-umd'
```
Comment (thornbill) [inline webpack.common.js:105]: Does this not need included as a dev dependency?

Comment (dmitrylyzo) [reply webpack.common.js:105]: :+1:  It is safer to add

**Review by thornbill — APPROVED** [review]

**Comment by oddstr13** [comment]
I'm testing with this PR rebased on top of master.

Testing is done on the following TV (`ares-device-info`):
```
modelName : 49UH603V-ZE
sdkVersion : 3.3.0
firmwareVersion : 05.30.65
boardType : K2L_DVB_EU
```

## Video
Playback seems flawless, but I don't have a extensive codec test library set up.

## Music
No OSD.

Attempting to play an unsupported track just stops playback.
```
Requesting url without automatic networking: http://10.79.1.0:8096/Sessions/Playing/Progress main.bundle.js:2
media element error: 4  htmlAudioPlayer-plugin.bundle.js:1
playbackmanager playback error type: medianotsupported main.bundle.js:2
Requesting url without automatic networking: http://10.79.1.0:8096/Sessions/Playing/Stopped main.bundle.js:2
```

## PDF
Forever loading on a white background;
```
Warning: Setting up fake worker. 2986.bundle.js:2
Uncaught SyntaxError: Unexpected token ( pdf.worker.js:2
Unhandled promise rejection Error: Setting up fake worker failed: "Cannot read property 'WorkerMessageHandler' of undefined". {stack: (...), message: "Setting up fake worker failed: "Cannot read property 'WorkerMessageHandler' of undefined"."} main.bundle.js:2
```

## epub
Loading for a while, then just a white screen. Looks like it's supposed to be displaying the cover image – I had the same issue in the browser, but that was fixed a while back. No javascript errors;
image

## CBZ
Infinite loading on dark background. The file does get fetched (if it can fit in cache, otherwise the app is forcefully restarted)
```
Uncaught SyntaxError: Use of const in strict mode. worker-bundle.js:1
```

<details>
<summary>
This error is observed with the CBZ player in Firefox
</summary>

```js
20:45:39.522 Uncaught (in promise) TypeError: URL.createObjectURL: Argument 1 is not valid for any of the 1-argument overloads.
    e http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    d http://1070l:8096/web/main.bundle.js:2
    _invoke http://1070l:8096/web/main.bundle.js:2
    k http://1070l:8096/web/main.bundle.js:2
    E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*E http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    a http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    P http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    P http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    value http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    value http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    promise callback*value/< http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    value http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    value http://1070l:8096/web/comicsPlayer-plugin.bundle.js:1
    J http://1070l:8096/web/main.bundle.js:2
    J http://1070l:8096/web/main.bundle.js:2
    play http://1070l:8096/web/main.bundle.js:2
    promise callback*e/f.play/< http://1070l:8096/web/main.bundle.js:2
    promise callback*e/f.play http://1070l:8096/web/main.bundle.js:2
    u http://1070l:8096/web/main.bundle.js:2
    p http://1070l:8096/web/main.bundle.js:2
    TU http://1070l:8096/web/main.bundle.js:2
    attachedCallback http://1070l:8096/web/main.bundle.js:2
    l http://1070l:8096/web/main.bundle.js:2
    b http://1070l:8096/web/main.bundle.js:2
    t http://1070l:8096/web/main.bundle.js:2
    upgrade http://1070l:8096/web/main.bundle.js:2
    o http://1070l:8096/web/main.bundle.js:2
    c http://1070l:8096/web/main.bundle.js:2
    t http://1070l:8096/web/main.bundle.js:2
    n http://1070l:8096/web/main.bundle.js:2
    n http://1070l:8096/web/main.bundle.js:2
    t http://1070l:8096/web/main.bundle.js:2
    c http://1070l:8096/web/main.bundle.js:2
    a http://1070l:8096/web/main.bundle.js:2
    h http://1070l:8096/web/main.bundle.js:2
    h http://1070l:8096/web/main.bundle.js:2
    h http://1070l:8096/web/main.bundle.js:2
    g http://1070l:8096/web/main.bundle.js:2
    y http://1070l:8096/web/main.bundle.js:2
    o http://1070l:8096/web/main.bundle.js:2
    forDocumentTree http://1070l:8096/web/main.bundle.js:2
    upgradeDocumentTree http://1070l:8096/web/main.bundle.js:2
```

</details>

**Comment by dmitrylyzo** [comment]
> No OSD.

On TV, music is controlled by remote page - `note` sign at the top menu.

**Comment by oddstr13** [comment]
> > No OSD.
> 
> On TV, music is controlled by remote page - `note` sign at the top menu.

Yes, I did find that. Kinda expected some sort of HUD still however – but that's irrelevant to this PR then.

**Review by joshuaboniface — APPROVED** [review]

**Pull request merged**